text/entities/0264-resource-and-entities.md

```diff
@@ -0,0 +1,841 @@
+identify conflicts against the set of attributes that make up an Entity.
+
+The merge rules defined here give precedent for the collector to generically interact with "type", "identifying attributes" and "descriptive attributes"
+rather than hard-coded rules that have to understand the nuance of when `host.id` influences `host.name`, e.g.
+
+### Users are expected to provide / prioritize "detectors" and determine which entity is "producing" or most-important for a signal
+
+The Resource Manager allows users to configure priority of Entity Detectors.
+
+### For an SDK - ALL telemetry should be associated with the same set of entities (resource labels)
+
+Resource Manager is responsible for resolving entities into a cohesive Resource that meets the same demands as Resource today.
+
+## Open Questions
+
+The following remain open questions:
+
+### How to attach Entity "bundle" information in Resource?
+
+The protocol today requires a raw grab bag of Attributes on Resource. We cannot break this going forward.  However, Entities represent a new mechanism of "bundling" attributes on Resource and interacting with these bundles.  We do not want this to bloat the protocol, nor do we want it to cause oddities.
+
+Going forward, we have set of options:
+
+- Duplicate attributes in `Entity` section of Resource.
+- Reference attributes of Resource in entity.
+- Only identify Entity id and keep attribute<->entity association out of band.
+- Extend Attribute on Resource so that we can track the entity type per Key-Value (across any attribute in OTLP).
+
+The third option prevents generic code from interacting with Resource and Entity without understanding the model of each.  The first keeps all usage of entity simple at the expense of duplicating information and the middle is awkward to interact with from an OTLP usage perspective. The fourth is violates our stability policy for OTLP.
+
+### How to deal with Resource/Entities whose lifecycle does not match the SDK?
+
+This proposal motivates a Resource Provider in the SDK whose job could include managing changes in entity lifetimes, but does not account for how these changes would be broadcast across TracerProvider, LogProvider, MeterProvider, etc.  That would be addressed in a follow on OTEP.
+
+### How to deal with Prometheus Compatibility for non-SDK telemetry?
+
+Today, [Prometheus compatibility](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/compatibility/prometheus_and_openmetrics.md) relies on two key attributes in Resource: `service.name` and `service.instance.id`. These are not guaranteed to exist outside of OpenTelemetry SDK generation. While this question is not fully answered, we believe outlining identity in all resources within OpenTelemetry allows us to define a solution in the future while preserving compatibility with what works today.
```

Comment:
@jsuereth could you here mention the possible solution of synthesizing a single identifier, for use within Prometheus, from the set of identifying entity attributes? E.g. it would become the `target_info` `instance` label when converting OTel metrics to Prometheus format. You mentioned in Slack we could use a "consistent-hash".

Reply:
Absolutely!

Reply:
done.